### PR TITLE
Update primitive compatibility flags and get_function

### LINKED ISSUE #1128: Refactor compatible primitive flags
Following [this](https://github.com/FeatureLabs/featuretools/pull/1031/files/af852ea92cedd2f47732e47b9acc348e10fef508#diff-65ffde62fc03046613de064c794ed250) comment, we should update how we flag primitives as compatible with the various DataFrame libraries to make it more scalable.

### PR DESCRIPTION
**Pull request by frances-h**
Fixes #1128 and #1132
- Update primitive compatibility from separate flags to a list of additional compatible libraries
- Add argument to `get_function` for aggregation primitives to return the correct aggregation for the given library
- Remove `get_dask_aggregation` from aggregation primitives

### CODE REVIEW

File: featuretools/primitives/base/primitive_base.py [hunk 1]
```diff
@@ -31,10 +31,8 @@ class PrimitiveBase(object):
     base_of_exclude = None
     # (bool) If True will only make one feature per unique set of base features
     commutative = False
-    # (bool) If True, is compatible with Dask EntitySets
-    dask_compatible = False
-    # (bool) If True, is compatible with Koalas EntitySets
-    koalas_compatible = False
+    #: (list): Additional compatible libraries
+    compatibility = []
```
Comment (gsheni) [inline featuretools/primitives/base/primitive_base.py:35]: What about if a certain primitive requires a higher version of dask/koalas? Should be an attribute to optionally specify the dask/koalas version?

File: featuretools/primitives/standard/aggregation_primitives.py [hunk 2]
```diff
@@ -33,14 +33,13 @@ class Count(AggregationPrimitive):
...
 
-    def get_function(self):
-        return pd.Series.count
-
-    def get_dask_aggregation(self):
-        return 'count'
+    def get_function(self, agg_type='pandas'):
+        if agg_type == 'pandas':
+            return pd.Series.count
+        elif agg_type == 'dask' or agg_type == 'koalas':
```
Comment (gsheni) [inline featuretools/primitives/standard/aggregation_primitives.py:41]: Not a big deal but could do `agg_type in ['dask', 'koalas']`

File: featuretools/primitives/utils.py [hunk 3]
```diff
@@ -39,14 +39,14 @@ def get_transform_primitives():
 
 def list_primitives():
     trans_names, trans_primitives = _get_names_primitives(get_transform_primitives)
-    trans_dask = [primitive.dask_compatible for primitive in trans_primitives]
+    trans_dask = ['dask' in primitive.compatibility for primitive in trans_primitives]
     transform_df = pd.DataFrame({'name': trans_names,
```
Comment (gsheni) [inline featuretools/primitives/utils.py:43]: Should there be a koalas column too?
```python
trans_koalas= ['koalas' in primitive.compatibility for primitive in trans_primitives]
transform_df = pd.DataFrame({'name': trans_names,
                             'description': _get_descriptions(trans_primitives),
                             'koalas_compatible':trans_koalas,
                             'dask_compatible': trans_dask})
```

File: featuretools/primitives/utils.py [hunk 4]
```diff
@@ -39,14 +39,14 @@ def get_transform_primitives():
...
                                  'description': _get_descriptions(trans_primitives),
                                  'dask_compatible': trans_dask})
     transform_df['type'] = 'transform'
 
     agg_names, agg_primitives = _get_names_primitives(get_aggregation_primitives)
-    agg_dask = [primitive.dask_compatible for primitive in agg_primitives]
+    agg_dask = ['dask' in primitive.compatibility for primitive in agg_primitives]
```
Comment (gsheni) [inline featuretools/primitives/utils.py:49]: add koalas?
`trans_koalas = ['koalas' in primitive.compatibility for primitive in trans_primitives]`

File: featuretools/primitives/utils.py [hunk 5]
```diff
@@ -39,14 +39,14 @@ def get_transform_primitives():
```
Comment (gsheni) [inline featuretools/primitives/utils.py:39]: Should there be a comment above to say that `get_aggregation_primitives` will now return all primitives? Whether or not they are compatible with dask/koalas? I know some users take `get_aggregation_primitives` and put it directly into dfs...

File: featuretools/utils/gen_utils.py [hunk 6]
```diff
@@ -133,3 +134,9 @@ def is_instance(obj, modules, classnames):
         raise ValueError('Number of modules does not match number of classnames')
     to_check = tuple(getattr(mod, classname, mod) for mod, classname in zip(modules, classnames) if mod)
     return isinstance(obj, to_check)
+
+
+class Lib(Enum):
```
Comment (kmax12) [inline featuretools/utils/gen_utils.py:139]: for clarity, perhaps just write out `Library` instead of shortening to `Lib`

**Comment by rwedge** [comment]
Do we want pandas compatibility to be implicit for all primitives?

File: featuretools/synthesis/deep_feature_synthesis.py [hunk 7]
```diff
@@ -238,13 +239,17 @@ def __init__(self,
             primitive_options = {}
         all_primitives = self.trans_primitives + self.agg_primitives + \
             self.where_primitives + self.groupby_trans_primitives
+        if any(isinstance(entity.df, pd.DataFrame) for entity in self.es.entities):
+            if not all([Library.PANDAS in primitive.compatibility for primitive in all_primitives]):
+                bad_primitives = ", ".join([prim.name for prim in all_primitives if Library.PANDAS not in prim.compatibility])
+                raise ValueError('Selected primitives are incompatible with pandas EntitySets: {}'.format(bad_primitives))
```
Comment (tamargrey) [inline featuretools/synthesis/deep_feature_synthesis.py:245]: Small thing, but if you wanted, you could do
```suggestion
bad_primitives = ", ".join([prim.name for prim in all_primitives if Library.PANDAS not in prim.compatibility])
if bad_primitives:
    raise ValueError('Selected primitives are incompatible with pandas EntitySets: {}'.format(bad_primitives))
```

Comment (tamargrey) [reply featuretools/synthesis/deep_feature_synthesis.py:245]: The pattern could follow for the dask and koalas ones below

Comment (rwedge) [reply featuretools/synthesis/deep_feature_synthesis.py]: Or something like
```
bad_primitives = [prim.name for prim in all_primitives if library not in prim.compatibility]
if bad_primitives:
    msg = "Selected primitives are incompatible with {} EntitySets: {}"
    raise ValueError(msg.format(library, ", ".join(bad_primitives))

**Review by rwedge — APPROVED** [review]
Looks good

**Pull request merged**